ui_testing/testcases/basic_tests/test06_orders.py

```diff
@@ -285,49 +279,59 @@ def test007_order_search(self, small_letters):
     def test008_duplicate_main_order(self):
         """
         New: Orders with test units: Duplicate an order with test unit 1 copy
-
         LIMS-3270
         :return:
         """
-        # get the random main order data
-        main_order = self.order_page.get_random_main_order_with_sub_orders_data()
-        # select the order
-        self.order_page.click_check_box(main_order['row_element'])
+        # get random order
+        self.base_selenium.LOGGER.info('select random record')
+        orders, payload = self.orders_api.get_all_orders(limit=20)
+        data_before_duplicate_main_order = random.choice(orders['orders'])
+
+        order_no = data_before_duplicate_main_order['orderNo']
+        self.order_page.apply_filter_scenario(filter_element='orders:filter_order_no', filter_text=order_no,
+                                              field_type='text')
+
+        row = self.order_page.get_last_order_row()
+        self.orders_page.open_edit_page(row)
+        data_before_duplicate = self.order_page.get_suborder_data()
+        self.orders_page.get_orders_page()
+        self.order_page.apply_filter_scenario(filter_element='orders:filter_order_no', filter_text=order_no,
+                                              field_type='text')
+
+        row = self.order_page.get_last_order_row()
+        self.order_page.click_check_box(source=row)
+
         # duplicate the main order
         self.order_page.duplicate_main_order_from_table_overview()
         # get the new order data
         after_duplicate_order = self.order_page.get_suborder_data()
 
-        # ignore contact no since the table view doesn't have contact No.
-        for contact in after_duplicate_order['contacts']:
-            contact['no'] = None
-
-        for index in range(len(main_order['suborders'])):
+        for index in range(len(data_before_duplicate['suborders'])):
             # ignore analysis no. since it won't be created in the form until saving
-            main_order['suborders'][index]['analysis_no'] = ''
+            data_before_duplicate['suborders'][index]['analysis_no'] = ''
             # ignore testunit numbers since the table view only got the name
             for testunit in after_duplicate_order['suborders'][index]['testunits']:
                 testunit['no'] = None
```

Comment:
why do you update data like this?
All you need to get data before duplicate and this can be done using api
then duplicte and get new order number
filter by it
get order and suborder date and compare them with data before duplicate

Reply:
@ghada-37 I want to ignore the analysis number because I don't want it before the saving process you will find the analysis number not found , then ignore the test unit number because in the form I have test unit number but the in the active table no